app/src/test/java/com/example/benchproject/app/popular/movies/viewModel/PopularMoviesViewModelTest.kt

```diff
@@ -58,20 +57,22 @@ internal class PopularMoviesViewModelTest {
 
     @Test
     fun `when method get throwable from use case, loader visibility should be true`() = runTest {
+        val slot = slot<() -> Unit>()
         coEvery { getPopularMoviesUseCase() } returns badResponse
-        every { popularMoviesFragmentNavigator.errorSnackBar(errorMessage) } just Runs
+        every { popularMoviesFragmentNavigator.errorSnackBar(errorMessage, onAction = capture(slot)) } answers {}
         val tested = PopularMoviesViewModel(getPopularMoviesUseCase, popularMoviesFragmentNavigator)
 
         tested.isLoaderVisible.value shouldBe true
     }
 
     @Test
     fun `when method get throwable from use case, errorSnackBar should run`() = runTest {
+        val slot = slot<() -> Unit>()
```

Comment:
Name of the test should adjusted

Reply:
I changed it. I am not sure it's a good name for this test but I used this `when the method gets throwable from use case, errorSnackBar should retry request to API`. Since I have one bad request from API and to avoid an infinite loop I sent the second request and I got success to end the loop. Therefore I checked the retry option from the user